### DISCUSSION TITLE
Failed mapping

### DISCUSSION THREAD

**Discussion by aplasa85** [neuefische/questions#361] [Web]
Hi, 

I wanted to ask for some advice, as I am encountering some issues here. I have so far done my layout and wanted to add to the index page some content. In this case this would be the pictures of the tequila brands which I have added to the template. For this I have written a map function which for some reason does not work properly. Indeed, I tried to declare a variable in Tequila List, but was not able to have it working neither. For all purposes I want to operate within the local storage and not go for a mongo db and then fetch. I would appreciate if someone could give me a piece of advice about how to implement this endeavour and what I am doing wrong. Thank you all in advance. 

https://github.com/aplasa85/capstone-template-ffm-web-22-1 

<img width="1440" alt="Screenshot 2022-12-02 at 14 54 16" src="https://user-images.githubusercontent.com/107868311/205308492-5d62dff5-eb2e-4c3f-b0ec-b225f09f31c5.png">

**Comment by actyralf** [comment]
Hey Axel, there are several issues with your `map()` method:

- You have a syntax problem: please check your linter output and the placement of curly brackets. The correct syntax for a map would be
   ```
   myArray.map((item)=>{return <Component>{item.title}</Component>})
   ```
   (myArray, Component and item to be replaced with your actual variable names)
- What exactly should `handleProduct.TequilaList.id` be? What do you expect to be the content of this variable?
- The `<Product>` component needs a unique `key` property

**Comment by aplasa85** [comment]
Hey Ralf,

I will take a look at it asap. My intent is to display tequila bottles
which are in the database. Will try to rewrite it and see if it works then.
Thanks for your feedback.


Am Freitag, 2. Dezember 2022 schrieb actyralf ***@***.***>:
